lib/llvm-backend-tests/tests/compile.rs

```diff
@@ -1,40 +1,72 @@
+use wasmer_llvm_backend::{InkwellMemoryBuffer, InkwellModule, LLVMBackendConfig, LLVMCallbacks};
 use wasmer_llvm_backend_tests::{get_compiler, wat2wasm};
-use wasmer_runtime::imports;
-use wasmer_runtime_core::compile_with;
+use wasmer_runtime::{imports, CompilerConfig};
+use wasmer_runtime_core::{backend::BackendCompilerConfig, compile_with, compile_with_config};
+
+use std::cell::RefCell;
+use std::rc::Rc;
 
 #[test]
 fn crash_return_with_float_on_stack() {
     const MODULE: &str = r#"
 (module
-  (type (;0;) (func))
-  (type (;1;) (func (param f64) (result f64)))
+  (type (func))
+  (type (func (param f64) (result f64)))
   (func $_start (type 0))
   (func $fmod (type 1) (param f64) (result f64)
     local.get 0
-    f64.const 0x0p+0 (;=0;)
+    f64.const 0x0p+0
     f64.mul
-    return)
-)
+    return))
 "#;
     let wasm_binary = wat2wasm(MODULE.as_bytes()).expect("WAST not valid or malformed");
     let module = compile_with(&wasm_binary, &get_compiler()).unwrap();
     module.instantiate(&imports! {}).unwrap();
 }
 
+#[derive(Debug, Default)]
+pub struct RecordPreOptIR {
+    preopt_ir: String,
+}
+
+impl LLVMCallbacks for RecordPreOptIR {
+    fn preopt_ir_callback(&mut self, module: &InkwellModule) {
+        self.preopt_ir = module.print_to_string().to_string();
+    }
+
+    fn postopt_ir_callback(&mut self, _: &InkwellModule) {}
+
+    fn obj_memory_buffer_callback(&mut self, _: &InkwellMemoryBuffer) {}
```

Comment:
You can provide (empty) defaults in the trait itself.  It may also be good to use `unimplemented!()` here to catch mistakes.

Reply:
Don't want to use `unimplemented!()` since it's totally reasonable to implement only one or two of them. Done!